From 01ba84716b98307880cd509d7ea3a9846ac41f1d Mon Sep 17 00:00:00 2001
From: Matthew Toohey <contact@mtoohey.com>
Date: Tue, 27 Feb 2024 16:52:58 -0500
Subject: [PATCH] Increase fallback fonts limit

---
 kitty/fonts.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/kitty/fonts.c b/kitty/fonts.c
index aeac4397d..81b5ba500 100644
--- a/kitty/fonts.c
+++ b/kitty/fonts.c
@@ -478,7 +478,7 @@ iter_fallback_faces(FONTS_DATA_HANDLE fgh, ssize_t *idx) {
 
 static ssize_t
 load_fallback_font(FontGroup *fg, CPUCell *cell, bool bold, bool italic, bool emoji_presentation) {
-    if (fg->fallback_fonts_count > 100) { log_error("Too many fallback fonts"); return MISSING_FONT; }
+    if (fg->fallback_fonts_count > 500) { log_error("Too many fallback fonts"); return MISSING_FONT; }
     ssize_t f;
 
     if (bold) f = italic ? fg->bi_font_idx : fg->bold_font_idx;
-- 
2.43.1

