From 0abee3aaff4a47c6b9b97e4fbe964ea6016bcee9 Mon Sep 17 00:00:00 2001
From: Matthew Toohey <contact@mtoohey.com>
Date: Fri, 11 Aug 2023 23:55:24 -0400
Subject: [PATCH] Add E3 terminfo capability to properly clear scrollback

---
 kitty/terminfo.py       | 3 +++
 terminfo/kitty.terminfo | 1 +
 2 files changed, 4 insertions(+)

diff --git a/kitty/terminfo.py b/kitty/terminfo.py
index d9523b7bf..3eb5c6cac 100644
--- a/kitty/terminfo.py
+++ b/kitty/terminfo.py
@@ -144,6 +144,8 @@ def encode_keystring(keybytes: bytes) -> str:
     # Delete the specified number of lines
     'dl': r'\E[%p1%dM',
     'dl1': r'\E[M',
+    # Clear scrollback buffer
+    'E3': r'\E[3J',
     # Erase specified number of characters
     'ech': r'\E[%p1%dX',
     # Clear to end of screen
@@ -369,6 +371,7 @@ def encode_keystring(keybytes: bytes) -> str:
     'mh': 'dim',
     'DL': 'dl',
     'dl': 'dl1',
+    'E3': 'E3',
     'ec': 'ech',
     'cd': 'ed',
     'ce': 'el',
diff --git a/terminfo/kitty.terminfo b/terminfo/kitty.terminfo
index b3beda3eb..8ea8bfcf5 100644
--- a/terminfo/kitty.terminfo
+++ b/terminfo/kitty.terminfo
@@ -20,6 +20,7 @@ xterm-kitty|KovIdTTY,
 	BE=\E[?2004h,
 	Cr=\E]112\007,
 	Cs=\E]12;%p1%s\007,
+	E3=\E[3J,
 	Ms=\E]52;%p1%s;%p2%s\E\\,
 	PE=\E[201~,
 	PS=\E[200~,
-- 
2.41.0

